cli/tests/pcluster/validators/test_cluster_validators.py

```diff
@@ -2764,3 +2765,107 @@ def test_root_volume_encryption_consistency_validator(
+        ),
+        pytest.param(
+            [
+                SlurmQueue(
+                    name="queue_1",
+                    compute_resources=[SlurmComputeResource(name="compute_resource_1", instance_type="instance_type")],
+                    networking=SlurmQueueNetworking(subnet_ids=["subnet_1", "subnet_2"], assign_public_ip=False),
+                ),
+            ],
+            2,
+            [
+                {"SubnetId": "subnet_1", "MapPublicIpOnLaunch": False},
+                {"SubnetId": "subnet_2", "MapPublicIpOnLaunch": False},
+            ],
+            None,
+            id="Test with multi nic queue with neither assigned public ip nor subnet with public ip",
+        ),
+        pytest.param(
+            [
+                SlurmQueue(
+                    name="queue_1",
+                    compute_resources=[SlurmComputeResource(name="compute_resource_1", instance_type="instance_type")],
+                    networking=SlurmQueueNetworking(subnet_ids=["subnet_1", "subnet_2"], assign_public_ip=True),
+                ),
+            ],
+            2,
+            [
+                {"SubnetId": "subnet_1", "MapPublicIpOnLaunch": False},
+                {"SubnetId": "subnet_2", "MapPublicIpOnLaunch": False},
+            ],
+            ["The queue queue_1 contains multi nic resources, but has AssignPublicIp set to true."],
+            id="Test with multi nic queue with assigned public ip and no subnet with public ip",
+        ),
+        pytest.param(
+            [
+                SlurmQueue(
+                    name="queue_1",
+                    compute_resources=[SlurmComputeResource(name="compute_resource_1", instance_type="instance_type")],
+                    networking=SlurmQueueNetworking(subnet_ids=["subnet_1", "subnet_2"], assign_public_ip=False),
+                ),
+            ],
+            2,
+            [
+                {"SubnetId": "subnet_1", "MapPublicIpOnLaunch": True},
+                {"SubnetId": "subnet_2", "MapPublicIpOnLaunch": False},
+            ],
+            [
+                "The queue queue_1 contains multi nic resources, but its subnets ['subnet_1'] "
+                "automatically assign public IPs."
+            ],
+            id="Test with multi nic queue with no assigned public ip and subnet with public ip",
+        ),
+        pytest.param(
+            [
+                SlurmQueue(
+                    name="queue-1",
+                    compute_resources=[SlurmComputeResource(name="compute_resource_1", instance_type="instance_type")],
+                    networking=SlurmQueueNetworking(subnet_ids=["subnet_1"], assign_public_ip=True),
+                ),
+            ],
+            2,
+            [{"SubnetId": "subnet_1", "MapPublicIpOnLaunch": True}],
+            [
+                "The queue queue-1 contains multi nic resources, but has AssignPublicIp set to true.",
```

Comment:
note: you can use `.*` inside the string to skip the part of the sentence not relevant for the assertion.